src/protocol/sort/generate_permutation.rs

```diff
@@ -274,7 +276,7 @@ where
     assert_eq!(sort_keys.len(), num_bits as usize);
 
     let upgraded_sort_keys = m_ctx
-        .upgrade_vector(&MaliciousUpgradeInput, sort_keys[0].clone())
+        .upgrade_vector(&MaliciousUpgradeInput(0), sort_keys[0].clone())
```

Comment:
Not sure if there's a reason these functions take a reference to `sort_keys`. It looks like the callers have an owned Vec anyways, and the upgrade functions want to take the raw input by value, so it may be possible to change the argument type to this function and get rid of this clone and several like it.

Reply:
Callers own the vector but it is used in the next step after sorting. We use the sort keys to sort and then use them to apply sort on combined_match_keys_and_sidecar_data. 
Great to see upgrade_vector go away! The only reason I included (0) was to run it across bits in malicious_multi